### PR TITLE
coreutils: update conflicts, link with gmp

### PR DESCRIPTION
**Pull request by carlocab**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [ ] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Does your build pass `brew audit --strict <formula>` (after doing `brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

`basenc`, ~~`dir`, `dircolors`~~, `factor` ~~and `vdir`~~ are not available
on macOS. Checked on Mojave, Catalina, and Big Sur.

Also add a gmp dependency. This enables `factor` to work on arbitrarily
large integers. Debian, Fedora, Arch and MacPorts all build coreutils
with gmp, and it's a fairly lightweight dependency (~3 MB).

### CODE REVIEW

**Comment by carlocab** [comment]
Weird failure for `ponysay`. Let's see if it reproduces elsewhere.

**Comment by carlocab** [comment]
Ok, this reproduces consistently. I can't reproduce it locally, though...

**Comment by carlocab** [comment]
@iMichka how do you decide what goes into the `no_conflict` array on Linux?

I want to automate the creation of this by checking `PATH` with `which` and adding it if the path starts with `/bin`, `/usr/bin`, or `/usr/sbin`. However, based on the Linux Docker container it seems like this would lead to fewer things in the `no_conflict` array.

**Comment by iMichka** [comment]
> how do you decide what goes into the no_conflict array on Linux?

I copied the list over from linuxbrew-core; might need to check with git blame there. I did not decide what went into it. @sjackman @dawidd6 might have an opinion.

**Comment by sjackman** [comment]
> how do you decide what goes into the no_conflict array on Linux?

Ignore the files on the host system, because that may vary from system to system. I would put everything in there that does not conflict with another Homebrew package, where that other package is the preferred provider of that file.

File: Formula/coreutils.rb [hunk 1]
```diff
@@ -49,16 +51,29 @@ def install
...
+      ]
+    end
+
     # Symlink all commands into libexec/gnubin without the 'g' prefix
     coreutils_filenames(bin).each do |cmd|
       (libexec/"gnubin").install_symlink bin/"g#{cmd}" => cmd
+
+      # Find non-conflicting commands on macOS
+      which_cmd = which(cmd)
+      no_conflict << cmd if OS.mac? && (which_cmd.nil? || !which_cmd.to_s.start_with?(%r{(/usr)?/s?bin}))
```
Comment (Bo98) [inline Formula/coreutils.rb:76]: This will cause a regression: https://github.com/Homebrew/homebrew-core/commit/1b198c839ba376943aa51988ba0a84570214e121

**Review by SMillerDev — APPROVED** [review]

**Pull request closed without merging**